src/main/java/catbot/CatBotCommandPatterns.java

```diff
@@ -0,0 +1,63 @@
+package catbot;
+
+import catbot.internal.CommandPattern;
+import catbot.internal.CommandPatternGenerator;
+import catbot.internal.NamedParameterMap;
+
+import java.util.function.Consumer;
+
+public class CatBotCommandPatterns {
```

Comment:
Maybe you can consider adding header comments/javadocs? For all your other classes as well

Reply:
Yup! I'll get around to it. _Eventually_. :')